internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -120,6 +120,37 @@ func (k *KafkaManager) Reconcile() []error {
 		return encounteredErrors
 	}
 
+	for _, kafka := range kafkas {
+		if !kafka.CanBeSuspended() {
+			// this kafka is not in a state that can be suspended
+			continue
+		}
+
+		expired, residualLifespan := kafka.IsExpired()
+		if expired {
+			// Expired kafkas will be deleted elsewhere
+			continue
+		}
+
+		// get the billing model
+		bm, err := k.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+		if err != nil {
+			wrappedError := errors.Wrap(err, "failed to suspend expired Kafka instances")
+			encounteredErrors = append(encounteredErrors, wrappedError)
+			continue
+		}
+
+		if !residualLifespan.Infinite() && residualLifespan.Days() < float64(bm.GracePeriodDays) {
```

Comment:
```suggestion
		if !residualLifespan.Infinite() && residualLifespan.IsInGracePeriod(bm.GracePeriodDays) {
```

What do you think of having a `IsInGracePeriod()` method?

Reply:
The residual life span object has no concept of `grace period`: it just returns the lifespan of the Kafka instance.
The concept of 'grace period' exists in the caller. However I could add a `LessThanOrEqual(days int) bool` method and make the code:
```go
residualLifespan.LessThanOrEqual(bm.GracePeriodDays)
```

Reply:
done

Reply:
Thanks